runtime/lua/vim/subprocess.lua

```diff
@@ -0,0 +1,153 @@
+local uv = vim.loop
+
+local function handle_input(pipe, input)
+  if type(input) == 'table' then
+    for _, v in ipairs(input) do
+      pipe:write(v)
+      pipe:write('\n')
+    end
+  elseif type(input) == 'string' then
+    pipe:write(input)
+  end
+
+  -- Shutdown the write side of the duplex stream and then close the pipe.
+  -- Note shutdown will wait for all the pending write requests to complete
+  pipe:shutdown(function()
+    if pipe then
+      pipe:close()
+    end
+  end)
```

Comment:
Hmm, well that suggests a bug in luv/libuv since the documentation exactly says:

>              Shutdown the outgoing (write) side of a duplex stream. It
>              waits for pending write requests to complete. The callback is
>              called after shutdown is complete.

But I guess there's no harm in just doing:

```lua
    stdin:write('', function()
      stdin:shutdown(function()
        stdin:close()
      end)
```

if you know that to work.